app/src/main/java/com/example/atheneum/activities/BookInfoActivity.java

```diff
@@ -198,13 +286,50 @@ public void onChanged(@Nullable Photo photo) {
             @Override
             public void onChanged(final @Nullable Book book) {
                 if (book != null) {
+
                     BookInfoActivity.this.book = book;
 
                     textTitle.setText(book.getTitle());
                     textAuthor.setText(book.getAuthor());
                     textIsbn.setText(String.valueOf(book.getIsbn()));
                     textDesc.setText(book.getDescription());
                     textStatus.setText(String.valueOf(book.getStatus()));
+//                    status = String.valueOf(book.getStatus());
+
+                    if (book.getStatus().equals(Book.Status.ACCEPTED) || book.getStatus().equals(Book.Status.BORROWED)) {
+                        Log.i(TAG, "***//scan button visible");
+                        scanBtn.setVisibility(View.VISIBLE);
+                        scanBtn.setOnClickListener(new View.OnClickListener() {
+                            @Override
+                            public void onClick(View v) {
+                                TransactionViewModelFactory factory = new TransactionViewModelFactory(book.getBookID());
+                                TransactionViewModel transactionViewModel = ViewModelProviders.of(BookInfoActivity.this, factory).get(TransactionViewModel.class);
+                                final LiveData<Transaction> transactionLiveData = transactionViewModel.getTransactionLiveData();
+
+                                transactionLiveData.observe(BookInfoActivity.this, new Observer<Transaction>() {
+                                    @Override
+                                    public void onChanged(@Nullable Transaction transaction) {
+                                        Log.i(TAG, "***// transaction getBScan: " + transaction.getBScan());
+                                        Log.i(TAG, "***// transaction getOScan: " + transaction.getOScan());
+                                        if (!transaction.getBScan()){
```

Comment:
You need to add null checks for the transaction here. Even though you don't expect the transaction to be null, if it ever happens the app will crash since you are trying to do tansaction.getBScan() on a null object reference. An easy fix to this is to add an if condition before this that checks that if transaction is equal to null, to log a warning.

Reply:
Ie. do Log.w(TAG, "transactin is null!")